### DISCUSSION TITLE
Is there any way to rollback if egde function failed?

### DISCUSSION THREAD

**Discussion by dnguyenfs** [supabase/supabase#8899] [Questions]
I have an example 
when create new location in location database. I also trigger to create new stripe customer and attach to it. But the problem is all I have to do is using edge function to secure my stripe keys. but what if my stripe function failed. it still create a location without customer_id. 

Is there anyway I can make it as a transaction?

**Comment by GaryAustin1** [comment]
If you get back successful/error status from the edge function you can raise an exception in the trigger function and it will abort the insert.
  
https://www.postgresql.org/docs/current/plpgsql-errors-and-messages.html

https://www.postgresql.org/docs/current/plpgsql-trigger.html
`The return value of a row-level trigger fired AFTER or a statement-level trigger fired BEFORE or AFTER is always ignored; it might as well be null. However, any of these types of triggers might still abort the entire operation by raising an error.`
